src/agent/border_agent.cpp

```diff
@@ -277,158 +339,57 @@ void BorderAgent::StopPublishService(void)
     otbrLog(OTBR_LOG_INFO, "Stop publishing service");
 }
 
-void BorderAgent::SetNetworkName(const char *aNetworkName)
+void BorderAgent::HandleThreadStateChanged(otChangedFlags aFlags)
 {
-    strcpy_safe(mNetworkName, sizeof(mNetworkName), aNetworkName);
+    VerifyOrExit(IsPskcInitialized() && mPublisher != nullptr);
```

Comment:
Will this catch PSKc change?

Reply:
We wanted to stop publishing the `meshcop` service if `PSKc` is set to all-zeros, right?

Reply:
yes, and recover when PSKc is set.

Reply:
okay. but we cannot simply stop the mDNS publisher now, because the advertising proxy should continue working when `PSKc` is zeroed. I will revise to unpublish the `_meshcop._udp` service instead.